deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJob.java

```diff
@@ -441,15 +441,14 @@ private void ingestFolder(ContentContainerObject parent, Resource parentResc, Re
 
```

Comment:
I think `ingestFileObject` could still be problematic since its happening inside of the transaction for a work. If it fails after a few files the count would be off. Handling this might be a bit tricky, you might have to keep a running count of the number of file objects added rather than committing to redis immediately, and then include that count in the addClicks and incrIngestedObjects when the transaction ends for the work object. It'd definitely be good to have a test to ensure the counts come out right in a failure case. Is there an existing test that might be used for this?

Reply:
Wouldn't this cover that case?
```java
if (skipResumed(childResc)) {
     return;
}
```

There is this test https://github.com/UNC-Libraries/Carolina-Digital-Repository/blob/fcrepo4/deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJobIT.java#L444. Maybe something like it would work?

Reply:
If the work transaction got rolled back, then the file wouldn't be in fedora, so we'd need to reingest the object. skipResumed is checking against objects in fedora, so it wouldn't prevent the fileObject from getting ingested and the counter from getting incremented.

I don't see a test that verifies resuming a work deposit with multiple files partway through, but you could make a new test from the one you linked, adding 2 or 3 files to the work and having it fail on one of the files, then fix the error and re-run it, then make sure the count is correct